context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx

```diff
@@ -0,0 +1,108 @@
+import MenuList from '@material-ui/core/MenuList';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import Link from '@material-ui/core/Link';
+import { encodeConfInUrl } from 'vitessce';
+
+import { SecondaryBackgroundTooltip } from 'js/shared-styles/tooltips';
+import useVisualizationStore from 'js/stores/useVisualizationStore';
+import { WhiteBackgroundIconButton } from 'js/shared-styles/buttons';
+
+import { StyledLinkIcon, StyledTypography, StyledEmailIcon } from './style';
+import 'vitessce/dist/es/production/static/css/index.css';
+
+const DEFAULT_LONG_URL_WARNING =
+  'Warning: this is a long URL which may be incompatible or load slowly with some browsers.';
+
+const DEFAULT_EMAIL_MESSAGE = 'Here is an interesting dataset I found in the HuBMAP Data Portal:';
+
+const visualizationStoreSelector = (state) => ({
+  vitessceConfig: state.vitessceConfig,
+  setOnCopyUrlWarning: state.setOnCopyUrlWarning,
+  setOnCopyUrlSnackbarOpen: state.setOnCopyUrlSnackbarOpen,
+});
+function VisualizationShareButton() {
+  const [open, toggle] = useReducer((v) => !v, false);
+  const anchorRef = useRef(null);
+  const { vitessceConfig, setOnCopyUrlWarning, setOnCopyUrlSnackbarOpen } = useVisualizationStore(
+    visualizationStoreSelector,
+  );
+
+  const copyToClipBoard = (conf) => {
+    const dummy = document.createElement('input');
+    document.body.appendChild(dummy);
+    const url = `${window.location.href.split('#')[0]}#${encodeConfInUrl({
+      conf,
+      onOverMaximumUrlLength: () => {
+        setOnCopyUrlWarning(DEFAULT_LONG_URL_WARNING);
+      },
+    })}`;
+    setOnCopyUrlSnackbarOpen(true);
+    dummy.setAttribute('value', url);
+    dummy.select();
+    document.execCommand('copy');
+    document.body.removeChild(dummy);
+  };
+  const emailUrl = (conf) => {
+    let longUrlWarning = '';
+    const url = `${window.location.href.split('#')[0]}#${encodeConfInUrl({
+      conf,
+      onOverMaximumUrlLength: () => {
+        longUrlWarning = DEFAULT_LONG_URL_WARNING;
+      },
+    })}`;
+    // We need to encode the URL so its parameters do not conflict with mailto's.
+    const encodedUrl = encodeURIComponent(url);
+    const mailtoLink = `mailto:?body=${longUrlWarning}%0D%0A%0D%0A${DEFAULT_EMAIL_MESSAGE} ${encodedUrl}`;
+    window.location.href = mailtoLink;
+  };
+
+  return (
+    <>
+      <SecondaryBackgroundTooltip title="Share Visualization">
+        <WhiteBackgroundIconButton ref={anchorRef} onClick={toggle}>
+          <ShareIcon color="primary" />
+        </WhiteBackgroundIconButton>
+      </SecondaryBackgroundTooltip>
+      <Popper open={open} anchorEl={anchorRef.current} placement="bottom-start" style={{ zIndex: 50 }}>
+        <Paper style={{ maxHeight: 200, overflow: 'auto' }}>
+          <ClickAwayListener onClickAway={toggle}>
+            <MenuList id="preview-options">
```

Comment:
This isn't keyboard accessible. I think there are a few props you can pass to the mui menu components to handle this.

Reply:
Discussed offline - will be resolved in a separate PR.

Reply:
Some of our other menus which follow this pattern aren't keyboard accessible either, so this isn't an issue specific to this PR. I'll just file an issue and do this later. Thanks!